mbed-os-env-cmake/Dockerfile

```diff
@@ -7,54 +7,6 @@ FROM ubuntu:20.04
 ARG WORKDIR=/root
 
 # ------------------------------------------------------------------------------
-# Install tools via apt
-ENV DEBIAN_FRONTEND=noninteractive
-RUN apt -y update && \
-    apt -y install git \
-    wget \
-    python3 \
-    python3-dev \
-    python3-setuptools \
-    python3-usb \
-    python3-pip \
-    software-properties-common \
-    build-essential \
-    cmake \
-    astyle \
-    mercurial \
-    ninja-build \
-    libssl-dev \
-    && apt clean && rm -rf /var/lib/apt/lists
-
-
-# ------------------------------------------------------------------------------
-# Install Python modules (which are not included in requirements.txt)
-RUN pip3 install -U \
-    mbed-cli \
-    cmake \
-    mbed-tools
-
-# Set up mbed environment
-WORKDIR /root/
-RUN wget https://github.com/ARMmbed/mbed-os/raw/master/requirements.txt && \
-    pip3 install -r requirements.txt
-
-# ------------------------------------------------------------------------------
-# Install arm-none-eabi-gcc
-WORKDIR /root/
-RUN wget -q https://developer.arm.com/-/media/Files/downloads/gnu-rm/9-2019q4/RC2.1/gcc-arm-none-eabi-9-2019-q4-major-x86_64-linux.tar.bz2 && \
-    tar -xjf gcc-arm-none-eabi-9-2019-q4-major-x86_64-linux.tar.bz2 && \
-    rm gcc-arm-none-eabi-9-2019-q4-major-x86_64-linux.tar.bz2
-ENV PATH="/root/gcc-arm-none-eabi-9-2019-q4-major/bin:${PATH}"
-
-
-# ------------------------------------------------------------------------------
-# Display and save environment settings
-RUN python3 --version | tee env_settings && \
-    arm-none-eabi-gcc --version | tee -a env_settings && \
-    (echo -n 'mbed-cli ' && mbed --version) | tee -a env_settings && \
-    (echo -n 'mbed-greentea ' && mbedgt --version) | tee -a env_settings && \
-    (echo -n 'mbed-host-tests ' && mbedhtrun --version) | tee -a env_settings
-
-
-WORKDIR /root
+# Bootstrap
+ADD bootstrap.sh /root
+RUN sh /root/bootstrap.sh
```

Comment:
I really dislike the change to a single command, a change to `bootstrap.sh` triggers a full rebuild and it is **slow** with multiarch.  Whereas, with various `RUN`, actions made before the modification are cached.

Reply:
I am agree with @pan- here, change it to a single command makes the huge docker image (1.4GB I think) into a single layer, every change will triggers a full rebuilding of whole layer. And this impair the benefit of the layer caching, and make the upload/download a slower process

Reply:
Another point is if this bootstrap scripts calling other scripts make changes to the docker image contents, the it would be unclear what it is in the docker image. 
One of major benefit of  Dockerfile is : It is self-descriptive, by reading the Dockerfile, you know what it is inside image

Reply:
The initial idea was to share one script for Docker and others (e.g. Vagrant), but I agree that performance (caching) should be prioritised.

It would be good to have this part reverted, while having other improvement (e.g. installing CMake via PPA) merged soon, so our CI (mainly Travis) can benefit from that.

Reply:
Some Travis scripts currently still fetch the raw Ubuntu image and install stuff manually at the moment

Reply:
(By "reverted" I mean a separate PR - no need to abandon any work we've done here)